packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1052,18 +1065,26 @@ export class EthModule {
     return new BN(block.header.number);
   }
 
-  private _extractBlock(blockTag: OptionalBlockTag): BN {
+  private async _extractBlock(blockTag: OptionalBlockTag): Promise<BN> {
+    if (BN.isBN(blockTag)) {
+      return blockTag;
+    }
+
+    if (Buffer.isBuffer(blockTag)) {
+      const block = await this._node.getBlockByHash(blockTag);
+      return block === undefined ? LATEST_BLOCK : new BN(block.header.number);
```

Comment:
Won't this mean that sending a non-existent block hash will be interpreted as `latest`?

Reply:
True. This could be changed to throw an exception instead. This function is currently only used for logs though. I don't h ave a strong opinion about this.

Reply:
I think the eip says that a resource not found error should be returned. It's a bit ambiguous on this front though.

Reply:
Yeah It's not the most detailed EIP I have to say. It's also quite ambiguous in terms of input validation in general. I think this implementation should be fine though. I will update the PR to throw an error.